File: driftcoach/outputs/insight.py
from dataclasses import dataclass
from typing import List

from driftcoach.core.derived_fact import DerivedFact


@dataclass(frozen=True)
class Insight:
    subject: str
    claim: str
    derived_facts: List[DerivedFact]
    confidence: float
    failure_conditions: List[str]

File: frontend/types/insight.ts
export type DerivedFact = {
  factType: string;
  value: number;
  baseline?: number | null;
  sampleSize: number;
  metadata?: Record<string, unknown>;
};

export type Insight = {
  subject: string;
  claim: string;
  derivedFacts: DerivedFact[];
  confidence: number; // 0-1
  failureConditions: string[];
  explanation?: string; // LLM-rendered text (frontend layer), optional
};
